odoo/api.py

```diff
@@ -631,25 +631,47 @@ def __call__(self, cr=None, user=None, context=None, su=None):
         return Environment(cr, uid, context, su, self.uid_origin)
 
-    def ref(self, xml_id, raise_if_not_found=True):
-        """ Return the record corresponding to the given ``xml_id``.
-
-        :param str xml_id: record xml_id, under the format ``<module.id>``
-        :param bool raise_if_not_found: whether the method should raise if record is not found
-        :returns: Found record or None
-        :raise ValueError: if record wasn't found and ``raise_if_not_found`` is True
+    def ref(self, *xml_ids: str, raise_if_not_found: bool = True) -> BaseModel | None:
+        """ Return the records corresponding to the given ``xml_ids``.
+
+        :param str xml_ids: record xml_ids, under the format ``<module>.<record_id>``
+        :param bool raise_if_not_found: whether the method should raise if any ``xml_id`` isn't found.
+        :returns: Found records or None
+        :raise ValueError: if any record wasn't and ``raise_if_not_found`` is True,
+            or if the user mixed ``xml_ids`` that belong to different models.
+
+        .. note::
+            If ``raise_if_not_found`` is ``False`` and there are multiple
+            ``xml_ids`` as input, then it won't be possible to determine which
+            specific records weren't found by only looking at the recordset.
         """
-        res_model, res_id = self['ir.model.data']._xmlid_to_res_model_res_id(
-            xml_id, raise_if_not_found=raise_if_not_found
-        )
 
-        if res_model and res_id:
-            record = self[res_model].browse(res_id)
-            if record.exists():
-                return record
-            if raise_if_not_found:
-                raise ValueError('No record found for unique ID %s. It may have been deleted.' % (xml_id))
-        return None
+        id_xmlid = {}
+        first_model_name, first_xml_id = None, None
+        for xml_id in xml_ids:
+            res_model, res_id = self['ir.model.data']._xmlid_to_res_model_res_id(
+                xml_id, raise_if_not_found=raise_if_not_found
+            )
+            if res_model and res_id:
+                if not first_model_name:
+                    first_model_name = res_model
+                    first_xml_id = xml_id
+                elif first_model_name != res_model:
+                    raise ValueError(
+                        f"'{first_xml_id}' and '{xml_id}' xmlids belong to different models."
+                    )
+                id_xmlid[res_id] = xml_id
+        if not id_xmlid:
+            return None
+
+        records = self[first_model_name].browse(id_xmlid)
+        existing = records.exists()
+        if raise_if_not_found and (missing := records - existing):
+            missing_ids = ", ".join(id_xmlid[record.id] for record in missing)
+            raise ValueError(
+                f"No records found for unique ID(s) {missing_ids}. They may have been deleted."
+            )
+        return existing or None
```

Comment:
the design of this API is OK for me
but we have to be very carefully in the futrue
`record1, record2 = self.env.ref('record1_xml_id', 'record2_xml_id')`
is only equal to 
`record1 = self.env.ref('record1_xml_id')`
`record2 = self.env.ref('record2_xml_id')`
when you know `record1` and `record2` will never be the same reocrd

Reply:
Why? Where does the de-duplication happen?

Reply:
Yes, this syntax is actually bad per-se, given that it's a recordset and there is the `exists`. 
I wouldn't use it if it was a `self.env[<model>].search([], limit=2)` unless the length is checked first, right?

Reply:
The de-duplication happens in `id_xmlid[res_id] = xml_id` where the latter one over-write the previous one, since they have the same res_id

Reply:
To be true, there's no reason to actively deduplicate 🤔
I will just build a list of `ids` as I find them and `browse` that.

Even though... we are using one model - who ever is going to use write code that has two xml_ids that point to the same record, and then put them in different variables like that... ? Very unlikely, to me. 🤔

Reply:
Done.